bin/describe.ml

```diff
@@ -612,45 +623,47 @@ module External_lib_deps = struct
       >>| Preprocess.Per_module.pps
     in
     Memo.parallel_map
-      ~f:(fun (_, name) ->
-        let+ is_external = is_external db name in
-        if is_external then Some { name; kind = Kind.Required } else None)
+      ~f:(fun (_, name) -> resolve_lib db name Kind.Required)
       pps
-    >>| List.filter_opt
 
-  let external_resolve db name kind =
-    let open Memo.O in
-    let+ is_external = is_external db name in
-    if is_external then Some { name; kind } else None
-
-  let external_lib_deps db lib_deps =
+  let resolve_lib_deps db lib_deps =
     let open Memo.O in
     lib_deps
     |> Memo.parallel_map ~f:(fun lib ->
            match lib with
-           | Lib_dep.Direct (_, name) | Lib_dep.Re_export (_, name) -> (
-             let+ v = external_resolve db name Kind.Required in
-             match v with
-             | Some x -> [ x ]
-             | None -> [])
+           | Lib_dep.Direct (_, name) | Lib_dep.Re_export (_, name) ->
+             let+ v = resolve_lib db name Kind.Required in
+             [ v ]
            | Lib_dep.Select select ->
              select.choices
              |> Memo.parallel_map ~f:(fun (choice : Lib_dep.Select.Choice.t) ->
                     Lib_name.Set.to_string_list choice.required
                     @ Lib_name.Set.to_string_list choice.forbidden
                     |> Memo.parallel_map ~f:(fun name ->
                            let name = Lib_name.of_string name in
-                           external_resolve db name Kind.Optional)
-                    >>| List.filter_opt)
+                           resolve_lib db name Kind.Optional))
              >>| List.concat)
     >>| List.concat
 
-  let external_libs db dir libraries preprocess names package kind =
+  let resolve_libs db dir libraries preprocess names package kind =
     let open Memo.O in
     let open Item in
-    let* lib_deps = external_lib_deps db libraries in
-    let+ lib_pps = external_lib_pps db preprocess in
-    Some { kind; dir; names; package; external_deps = lib_deps @ lib_pps }
+    let* lib_deps = resolve_lib_deps db libraries in
+    let+ lib_pps = resolve_lib_pps db preprocess in
+    let deps = lib_deps @ lib_pps in
+    let internal_deps =
+      deps
+      |> List.filter_map ~f:(function
+           | Local lib_dep -> Some lib_dep
+           | External _ -> None)
+    in
+    let external_deps =
```

Comment:
You can use `List.partition_map` to compute `internal_deps` and `external_deps` in a single step here